### PR TITLE
[FEATURE SERVERSIDE APPLY] Structure manager info in api

### PR DESCRIPTION
**Pull request by jennybuckley**
**What type of PR is this?**
/kind feature
/sig api-machinery
/priority important-soon
/assign @kwiesmueller @apelisse 

**What this PR does / why we need it**:
Change the managedFields map in the api to a list, to make it more like other kubernetes fields
Also change the specially formatted string "\<useragent\>-\<version\>@\<timestamp\>" to multiple fields in each list element.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields_test.go [hunk 1]
```diff
@@ -141,3 +152,38 @@ func TestRoundTripManagedFields(t *testing.T) {
...
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.managedFieldsEntry, func(t *testing.T) {
+			var unmarshaled metav1.ManagedFieldsEntry
+			if err := yaml.Unmarshal([]byte(test.managedFieldsEntry), &unmarshaled); err != nil {
+				t.Fatalf("did not expect yaml unmarshalling error but got: %v", err)
+			}
+			decoded, err := decodeManager(&unmarshaled)
```
Comment (kwiesmueller) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields_test.go]:
```suggestion
			decoded, err := DecodeManager(&unmarshaled)
```

**Review by kwiesmueller — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields_test.go (hunk 1 above)
Comment (jennybuckley) [reply staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/managedfields_test.go]: Thanks! I think I also need to update something in patchhandler_test

**Pull request merged**